Feature/deprecate set centroids

tovogt opened the pull request:
Replace and deprecate set-methods for the Centroids class.

chahank commented on climada/hazard/centroids/centr.py:
```diff
@@ -276,17 +276,29 @@ def set_raster_from_pix_bounds(self, xf_lat, xo_lon, d_lat, d_lon, n_lat,
+
+        Returns
+        -------
+        Centroids
```
```suggestion
        centr : climada.hazard.Centroids()
            Centroids defined by pixel border data.
```

tovogt commented:
Thanks for the review!

I merged #311 into this one, and addressed all of your comments. Note however, that I stick to the type `Centroids` in the docstrings because I think that `climada.hazard.Centroids` is too verbose and that it's incorrect to put parenthesis at the end of the type (compare, e.g., https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html or https://numpy.org/doc/stable/reference/generated/numpy.arange.html).

chahank commented on climada/hazard/centroids/centr.py:
```diff
@@ -276,17 +276,29 @@ def set_raster_from_pix_bounds(self, xf_lat, xo_lon, d_lat, d_lon, n_lat,
 
-    def set_raster_from_pnt_bounds(self, points_bounds, res, crs=DEF_CRS):
+    @classmethod
+    def from_pnt_bounds(cls, points_bounds, res, crs=DEF_CRS):
```
Fair point!

chahank commented:
Good point about the verbosity. Could you bring it up at the developer's meeting to inform all developers and agree on a standard? I will put it on the agenda.
